### ISSUE TITLE
Strophe requires from attribute in iq response

### ISSUE THREAD

**Issue by sualko** [strophe/strophejs#172]
As you can see in [src/core.js#L2043](https://github.com/strophe/strophejs/blob/master/src/core.js#L2043) Strophe requires that iq responses have a from attribute if you set a to attribute. This is conflicting to the [RFC6120](http://xmpp.org/rfcs/rfc6120.html#stanzas-semantics-iq) and therefore result in issues with Prosody 0.9.8. For example if you send a iq stanza to your own bar jid, prosody will send you the response without from attribute. A workaround is to remove the to attribute, but nevertheless this is a bug.

```
> <iq type="get" to="klaus@localhost" xmlns="jabber:client" id="97d:sendIQ"><vCard xmlns="vcard-temp"/></iq>

< <iq xmlns="jabber:client" id="97d:sendIQ" type="error" to="klaus@localhost/429"><error type="cancel"><item-not-found xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/></error></iq>
```

```
error: Got answer to IQ from wrong jid:null↵Expected jid: klaus@localhost↵undefined 
```

**Comment by jcbrand** [comment]
That code was added by @Gordin in this commit: https://github.com/strophe/strophejs/commit/6f16ad360cb7e7a5051421d62a68bae0a03bc0d3 and then refactored slightly here: https://github.com/strophe/strophejs/commit/42340f59f5375b3781c9874a871b69c434901223

The check doesn't really look necessary to me either, because the handler already is restricted to the same id of the requesting IQ, which appears to be the correct way of determining whether this is the right IQ response.

I think what might have happened is that @Gordin had issues with ID collisions. In other words, the ID generator wasn't unique enough, so he added another check to make sure that we were dealing with the right IQ response (I'm guessing here). We [recently updated the ID generator to generate UUIDs](https://github.com/strophe/strophejs/pull/144), so that problem, if it was indeed the problem, should be fixed now.

@Gordin Could you please explain your motivation for the changes above?

**Comment by jcbrand** [comment]
ping @Gordin 

@sualko: Would you mind making a pull request for this?

**Comment by Gordin** [comment]
@jcbrand I don't remember the exact reason, but I think your guess with the ID collisions was correct. I'm not sure any more If I encountered the problem or if I just added it as a sanity check though.

I guess if you wanted to keep the check in place the correct behaviour would be to set expectedFrom to ```null``` if 'to' is your own bare jid or your domain. (If I read [RFC6120#8.1.2.1](https://tools.ietf.org/html/rfc6120#section-8.1.2.1) right, that should cover the cases where the from attribute is allowed to be missing).

I'm not opposed to removing the check though, right know I can't think of a scenario where the check would make a difference if the IDs are unique enough.
